### PR TITLE
Rename pose_tracks to position

### LINKED ISSUE #100: Rename 'pose_tracks' to 'position'
Currently, movement `datasets` contain two data variables: `pose_tracks` and `confidence`. With #3, we are planning to also store some derived data variables - e.g. `velocity` and `acceleration` - within the same dataset. For naming consistency, we may want to rename `pose_tracks` to `position`. Almost everything in `movement` is 'tracks' anyway.

### PR DESCRIPTION
**Pull request by niksirbi**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

With the addition of kinematic data variables named `displacement`, `velocity`, `acceleration` etc., it's awkward to keep calling the variable containing poses `pose_tracks`. All of these variables are "tracks", in the sense that they are time-series. 

**What does this PR do?**

It renames the `xarray.DataArray` previously known as `pose_tracks` to `position`. This is more consistent with the naming of the other variables. I also did some additional renaming in comments/docstrings and in a few variables/functions/class names, to ensure that they are consistent with the new terminology.

Note that in many places in the docs and in comments/docstrings I decided to keep using the term "pose tracks". But this usage refers to the overall type of the data - i.e. tracks of poses over time - not the particular variable/attribute. This should become clearer once we address #69.

## References

Closes #100 

## How has this PR been tested?

This kind of change would be very painful without existing tests. Our nearly universal test coverage allowed me to catch the few instances I'd missed. I fixed them and the tests all pass now.

## Does this PR require an update to the documentation?

Yes, and I went through the docs and updated all instances I (and my IDEs) could find.
I've also updated an image that contained the term `pose_tracks` as the name of the corresponding `DataArray`.

## How to review

The diff of this PR will be impossible to review directly, as almost all files have been affected. Perhaps it's more critical to go over our public docs and double-check that the term is used consistently there.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by lochhh** [comment]
note: update poster examples

File: tests/test_unit/test_filtering.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 @pytest.fixture(scope="module")
 def sample_dataset():
-    """Return a dict containing a single- and a multi-animal sample dataset"""
+    """Return a single-animal sample dataset"""
```
Comment (lochhh) [inline tests/test_unit/test_filtering.py:17]:
```suggestion
    """Return a single-individual sample dataset"""
```
We use a mix of single- and multi-**animal/individual** throughout `movement`, and I think the use of "animal" in docstrings originated from DLC datasets, but in `movement` we use "individual". What do you think if we retain "animal" whenever it's used in non-`movement` context, e.g. "single-animal DeepLabCut project" and change the rest to "individual", e.g. `invalid_single_animal_csv_file` &rarr; `invalid_single_individual_csv_file` ?

File: tests/test_unit/test_save_poses.py
Comment (lochhh) [inline tests/test_unit/test_save_poses.py:1]: nitpicking: we still have `valid_pose_dataset` and `invalid_pose_dataset` which could be changed to `valid_poses_dataset` and `invalid_poses_dataset`, to match the change to `ValidPosesDataset`

File: tests/test_unit/test_validators.py [hunk 2]
```diff
@@ -27,46 +27,46 @@ class TestValidators:
...
+            "array_type": "multi_individual_array",
             "individual_names_expected_exception": pytest.raises(ValueError),
             "keypoint_names_expected_exception": pytest.raises(ValueError),
         },  # invalid input
     ]
 
-    @pytest.fixture(params=pose_tracks)
-    def pose_tracks_params(self, request):
+    @pytest.fixture(params=position_arrays)
+    def position_arrays_params(self, request):
```
Comment (lochhh) [inline tests/test_unit/test_validators.py:67]:
```suggestion
    def position_array_params(self, request):
```
as it's returning params for a single position array at a time

**Review by lochhh — APPROVED** [review]
LGTM and thanks @niksirbi! Made a couple of suggestions here and there but nothing major.

File: tests/test_unit/test_save_poses.py
Comment (niksirbi) [reply tests/test_unit/test_save_poses.py:1]: We love nitpicking! Will do.

File: movement/filtering.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def report_nan_values(ds: xr.Dataset, ds_label: str = "dataset"):
         nan_report += f"\n\tIndividual: {ind}"
         for kp in ds.keypoints.values:
             # Get the track for the current individual and keypoint
-            track_ = ds.pose_tracks.sel(individuals=ind, keypoints=kp)
+            track_ = ds.position.sel(individuals=ind, keypoints=kp)
             # A point is considered NaN if any of its space coordinates are NaN
             n_nans = track_.isnull().any(["space"]).sum(["time"]).item()
             n_points = track_.time.size
```
[the remark this answers is not in the document]
Comment (lochhh) [reply movement/filtering.py]: apologies this is the misaligned indentation!

File: tests/test_unit/test_filtering.py (hunk 1 above)
Comment (niksirbi) [reply tests/test_unit/test_filtering.py]: Good suggestion, will do.

**Comment by niksirbi** [comment]
Thanks for the very thorough review @lochhh! I've implemented your suggestions.

**Pull request merged**